pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/transformer/MetsXml.scala

```diff
@@ -134,6 +133,37 @@ case class MetsXml(root: Elem) {
     }
   }
 
+  /** Here we use the the items defined in the physicalStructMap to look up
+    * file IDs in the (normalised) fileObjects mapping
+    */
+  def fileReferencesMapping(bnumber: String): List[(String, FileReference)] =
```

Comment:
Might be missing something obvious here but why are we using these tuples/mappings rather than adding the `id` to the `FileReference` ADT?

Reply:
I don't think the physical ID is to do with a file reference (which has it's own ID), so didn't make sense to me to shove it on there :thinking: 

(nit: `FileReference` isn't an ADT I don't think? just a standard data type)

Reply:
Case classes are syntactic sugar for product types - they're just fancy tuples

Reply:
Actually the reason I referred to it as an ADT here is because recognising it as such made it look strange to me - the type `(String, FileReference)` is the product type, `String * (Whatever * Is * In * FileReference)` which associativity makes equivalent to `String * Whatever * Is * In * FileReference`, ie extending `FileReference`

Reply:
I get what you are saying about intrinsically being tuples, but here I think the choice is more to do with how we conceptualise and name things in relation to the logic of the application rather than underlying types and equivalence between them. Still interesting though. I generally use ADT here as shorthand to refer to sum types rather than product types, probably due to lack of any accepted Scala terminology for referring to a sealed trait with another of case class variants.